handler/handler.go

```diff
@@ -54,94 +54,48 @@ func (batchHandler BatchHandler) sendToES(ctx context.Context, esDestURL string,
 
 	log.Info(ctx, "bulk events into ES starts")
 	target := len(events)
-	documentList := make(map[string]models.SearchDataImportModel)
 
-	var bulkcreate []byte
+	var bulkupsert []byte
 	for _, event := range events {
-		if event.Title == "" {
-			log.Info(ctx, "No title for inbound event, no transformation possible")
-			continue // break here
-		}
-
-		documentList[event.Title] = *event
-		createBulkRequestBody, err := prepareEventForBulkRequestBody(ctx, event, "create")
+		upsertBulkRequestBody, err := prepareEventForBulkUpsertRequestBody(ctx, event)
 		if err != nil {
-			log.Error(ctx, "error in preparing the bulk for create", err, log.Data{
+			log.Error(ctx, "error in preparing the bulk for upsert", err, log.Data{
 				"event": *event,
 			})
 			continue
 		}
-		bulkcreate = append(bulkcreate, createBulkRequestBody...)
+		bulkupsert = append(bulkupsert, upsertBulkRequestBody...)
 	}
 
-	jsonCreateResponse, _, err := batchHandler.esClient.BulkUpdate(ctx, esDestIndex, esDestURL, bulkcreate)
+	jsonUpsertResponse, _, err := batchHandler.esClient.BulkUpdate(ctx, esDestIndex, esDestURL, bulkupsert)
 	if err != nil {
-		if jsonCreateResponse == nil {
-			log.Error(ctx, "error from elasticsearch while creating the event", err)
+		if jsonUpsertResponse == nil {
+			log.Error(ctx, "server error while upserting the event", err)
 			return err
 		}
-		log.Warn(ctx, "error in response from elasticsearch while creating the event", log.FormatErrors([]error{err}))
+		log.Warn(ctx, "error in response from elasticsearch while upserting the event", log.FormatErrors([]error{err}))
 	}
 
-	var bulkCreateRes models.EsBulkResponse
-	err = json.Unmarshal(jsonCreateResponse, &bulkCreateRes)
-	if err != nil {
-		log.Error(ctx, "error unmarshaling jsonCreateResponse", err)
-		return err
+	var bulkRes models.EsBulkResponse
+	if err := json.Unmarshal(jsonUpsertResponse, &bulkRes); err != nil {
+		log.Error(ctx, "error unmarshaling json", err)
 	}
-
-	if bulkCreateRes.Errors {
-		var bulkupdate []byte
-		for _, resCreateItem := range bulkCreateRes.Items {
-			if resCreateItem["create"].Status == 409 {
-				esEvent := documentList[resCreateItem["create"].ID]
-				updateBulkRequestBody, err := prepareEventForBulkRequestBody(ctx, &esEvent, "update")
-				if err != nil {
-					log.Error(ctx, "error in preparing the bulk for update", err)
-					continue
-				}
-				bulkupdate = append(bulkupdate, updateBulkRequestBody...)
-			} else if resCreateItem["create"].Status == 201 {
+	if bulkRes.Errors {
+		for _, resUpsertItem := range bulkRes.Items {
+			if resUpsertItem["create"].Status == 409 {
```

Comment:
The value `"create"` should be a constant. Then if that value changes in the future, then it is a matter of just changing the value of constant from the one place